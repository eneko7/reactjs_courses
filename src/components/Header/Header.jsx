import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from '../Logo';
import Search from './Search';
import SearchBy from './SearchBy';
import Button from '../Button';
import styles from './Header.scss';

class Header extends Component {
  searchHeandler = () => {
    const {
      searchRequest, sortBy, searchBy, fetchSearchMovies, history,
    } = this.props;
    fetchSearchMovies(searchRequest, searchBy, sortBy);
    if (searchRequest !== '') {
      history.push(`/search?q=${searchRequest}`);
    } else {
      history.push('');
    }
  };

  render() {
    return (
      <header className={styles.header}>
        <Logo />
        <div className={styles.header__find_your_movie}>
          <span className={styles.header__find_your_movie_span}>FIND YOUR MOVIE</span>
        </div>
        <Search />
        <div className={styles.header__search_controls}>
          <SearchBy />
          <Button classButton="header_button" text="search" onClick={this.searchHeandler} />
        </div>
      </header>
    );
  }
}

Header.defaultProps = {
  searchRequest: '',
};

Header.propTypes = {
  searchRequest: PropTypes.string,
  sortBy: PropTypes.string.isRequired,
  searchBy: PropTypes.string.isRequired,
  fetchSearchMovies: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default Header;
