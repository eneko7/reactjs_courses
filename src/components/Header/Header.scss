@import "../../assets/constants/constants.scss";

.header {
  height: 30vh;
  background-image: url('../../assets/images/netflix-image.jpg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
  z-index: 3;
  padding: 1% 15%;
  min-height: 260px;

  .header__find_your_movie {
    margin-top: 4%;
    margin-bottom: 2%;
    display: block;
    position: relative;
    z-index: 3;

    .header__find_your_movie_span {
      font-family: $fontFamilyBold;
      color: #fff;
      font-size: 25px;
    }
  }
}

.header::after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.6);
  z-index: 2;
}

.header__search_controls {
  position: inherit;
  z-index: inherit;
  display: flex;
  justify-content: space-between;
}

@media (max-width: $fulhdDesktop) {
  .header {
    padding: 1% $leftPaddingLaptop;
  }
}

@media (max-width: $laptop) {
  .header {
    padding: 1% $leftPaddingTablet;
  }
}

@media (max-width: $tablet) {
  .header {
    padding: 1% 5%;
    height: 32vh;
  }

  .header__search_controls {
    display: block;
  }
}
