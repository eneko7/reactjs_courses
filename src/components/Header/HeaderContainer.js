import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import Header from './Header';
import { getSearchBy, getSortBy, getSearchRequest } from '../../modules/filter/filterSelectors';
import { fetchSearchMovies } from '../../modules/movies/moviesActions';

const mapStateToProps = state => ({
  searchRequest: getSearchRequest(state),
  sortBy: getSortBy(state),
  searchBy: getSearchBy(state),
});

const mapDispatchToProps = {
  fetchSearchMovies,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));
