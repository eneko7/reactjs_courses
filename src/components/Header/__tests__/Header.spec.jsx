import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import Header from '../Header';

const renderer = new ShallowRenderer();
const data = {
  searchRequest: 'test',
  sortBy: 'release_date',
  searchBy: 'title',
};
const failedData = {
  searchRequest: '',
  sortBy: 'release_date',
  searchBy: 'title',
};
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  searchRequest: 'test',
  filter: {
    sortBy: 'release_date',
    searchBy: 'title',
  },
});
const failedStore = mockStore({
  searchRequest: '',
  filter: {
    sortBy: 'release_date',
    searchBy: 'title',
  },
});
jest.mock('../../SortInput', () => () => <div>test</div>);
const historyMock = { push: jest.fn() };

describe('Header', () => {
  it('renders correctly', () => {
    renderer.render(
      <Provider store={store}>
        <MemoryRouter>
          <Header {...data} fetchSearchMovies={() => ('Hello')} history={historyMock} />
        </MemoryRouter>
      </Provider>,
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('click render', () => {
    const component = TestRenderer.create(
      <Provider store={failedStore}>
        <MemoryRouter>
          <Header {...failedData} fetchSearchMovies={() => ('Hello')} history={historyMock} />
        </MemoryRouter>
      </Provider>,
    );
    component.root.findByProps({ className: 'header_button' }).props.onClick();
    expect(component).toMatchSnapshot();
  });

  it('click render', () => {
    const component = TestRenderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Header {...data} fetchSearchMovies={() => ('Hello')} history={historyMock} />
        </MemoryRouter>
      </Provider>,
    );
    component.root.findByProps({ className: 'header_button' }).props.onClick();
    expect(component).toMatchSnapshot();
  });
});
