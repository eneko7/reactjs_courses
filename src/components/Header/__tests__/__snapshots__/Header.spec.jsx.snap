// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header click render 1`] = `
<header
  className="header"
>
  <span
    className="logo"
  >
    NetflixRoulette
  </span>
  <div
    className="header__find_your_movie"
  >
    <span
      className="header__find_your_movie_span"
    >
      FIND YOUR MOVIE
    </span>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="search_input"
      name="search"
      onChange={[Function]}
      placeholder="Enter movie title"
      type="text"
      value=""
    />
    <span
      className="search_icon"
      onClick={[Function]}
      role="presentation"
    >
      â†©
    </span>
  </form>
  <div
    className="header__search_controls"
  >
    <div
      className="search_by_wrapper"
    >
      <span
        className="search_by_title"
      >
        SEARCH BY
      </span>
      <div>
        test
      </div>
      <div>
        test
      </div>
    </div>
    <button
      className="header_button"
      onClick={[Function]}
      type="button"
    >
      search
    </button>
  </div>
</header>
`;

exports[`Header renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "filter": Object {
          "searchBy": "title",
          "sortBy": "release_date",
        },
        "searchRequest": "test",
      },
    }
  }
>
  <Header
    fetchSearchMovies={[Function]}
    searchBy="title"
    searchRequest="test"
    sortBy="release_date"
  />
</Context.Provider>
`;
